src/v/pandaproxy/schema_registry/store.h

```diff
@@ -349,13 +349,9 @@ class store {
       seq_marker marker,
       const subject& sub,
       compatibility_level compatibility) {
-        auto sub_it = BOOST_OUTCOME_TRYX(
-          get_subject_iter(sub, include_deleted::no));
-
-        sub_it->second.written_at.push_back(marker);
-
-        // TODO(Ben): Check needs to be made here?
-        return std::exchange(sub_it->second.compatibility, compatibility)
+        auto& sub_entry = _subjects[sub];
```

Comment:
Deletion might break on subjects that have no versions -- I think `delete_subject_impermanent` assumes there is always at least one version (because `delete_subject_value` expects a version), but it probably doesn't need to, I'm not sure that version in the value is even used when applying the delete_subject message.

Reply:
Good spot. There's still a failing test because if there are no versions, deleting a version tries to delete the subject. I'll fix it and add commit.

Reply:
Done